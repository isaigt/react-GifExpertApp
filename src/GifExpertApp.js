import React,{ useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';
// import PropTypes from 'prop-types';

const GifExpertApp = ({ defaultCategorias=[] }) => {
    const [categorias, setCategorias] = useState( defaultCategorias );

    return (
        <>
            <h2 className="animate__animated animate__lightSpeedInLeft">GifExpertApp</h2>
            <AddCategory setCategorias={ setCategorias } />
            <hr />
            <ol>
                {
                    categorias.map( category => (
                        <GifGrid key={ category } categoria={ category } />
                    ))
                }
            </ol>
        </>
    );
}

// GifExpertApp.propTypes = {

// }

export default GifExpertApp;