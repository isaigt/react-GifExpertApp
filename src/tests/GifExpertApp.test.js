import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe('Pruebas del componente <GifExpertApp />', () => {
    test('Debe hacer match con el snapshot', () => {
        const wrapper = shallow( <GifExpertApp /> );
        expect( wrapper ).toMatchSnapshot();
    });

    test('debe de mostrar el numero correspondiente de GifGrids', () => {
        const categorias = ['Attack on titan', 'One Punch'];
        const wrapper = shallow( <GifExpertApp  defaultCategorias={ categorias } /> );

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( categorias.length );
    });
});