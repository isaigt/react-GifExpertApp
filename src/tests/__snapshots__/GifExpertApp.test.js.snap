// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente <GifExpertApp /> Debe hacer match con el snapshot 1`] = `
<Fragment>
  <h2
    className="animate__animated animate__lightSpeedInLeft"
  >
    GifExpertApp
  </h2>
  <AddCategory
    setCategorias={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas del componente <GifExpertApp /> debe de mostrar el numero correspondiente de GifGrids 1`] = `
<Fragment>
  <h2
    className="animate__animated animate__lightSpeedInLeft"
  >
    GifExpertApp
  </h2>
  <AddCategory
    setCategorias={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      categoria="Attack on titan"
      key="Attack on titan"
    />
    <GifGrid
      categoria="One Punch"
      key="One Punch"
    />
  </ol>
</Fragment>
`;
